ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -132,42 +132,59 @@ public ValidationResult<TransactionInvalidReason> validate(
     }
 
     if (transactionType == TransactionType.DELEGATE_CODE) {
-      if (isDelegateCodeEmpty(transaction)) {
-        return ValidationResult.invalid(
-            TransactionInvalidReason.EMPTY_CODE_DELEGATION,
-            "transaction code delegation transactions must have a non-empty code delegation list");
+      ValidationResult<TransactionInvalidReason> codeDelegationValidation =
+          validateCodeDelegation(transaction);
+      if (!codeDelegationValidation.isValid()) {
+        return codeDelegationValidation;
       }
+    }
+
+    return validateCostAndFee(transaction, baseFee, blobFee, transactionValidationParams);
+  }
 
-      final BigInteger halfCurveOrder = SignatureAlgorithmFactory.getInstance().getHalfCurveOrder();
-      final Optional<ValidationResult<TransactionInvalidReason>> validationResult =
-          transaction
-              .getCodeDelegationList()
-              .map(
-                  codeDelegations -> {
-                    for (CodeDelegation codeDelegation : codeDelegations) {
-                      if (codeDelegation.signature().getS().compareTo(halfCurveOrder) > 0) {
-                        return ValidationResult.invalid(
-                            TransactionInvalidReason.INVALID_SIGNATURE,
-                            "Invalid signature for code delegation. S value must be less or equal than the half curve order.");
-                      }
-
-                      if (codeDelegation.signature().getRecId() != 0
-                          && codeDelegation.signature().getRecId() != 1) {
-                        return ValidationResult.invalid(
-                            TransactionInvalidReason.INVALID_SIGNATURE,
-                            "Invalid signature for code delegation. RecId value must be 0 or 1.");
-                      }
+  private static ValidationResult<TransactionInvalidReason> validateCodeDelegation(
+      final Transaction transaction) {
+    if (isDelegateCodeEmpty(transaction)) {
+      return ValidationResult.invalid(
+          TransactionInvalidReason.EMPTY_CODE_DELEGATION,
+          "transaction code delegation transactions must have a non-empty code delegation list");
+    }
+
+    if (transaction.getTo().isEmpty()) {
```

Comment:
This is the new check to make sure that the `to` field is set. 

The rest is just a refactoring to move the validation of 7702 txs into their own method, because the main validation method became too long.